### ISSUE TITLE
Dead-code removal not working on return expressions

### ISSUE THREAD

**Issue by manucorporat** [rollup/rollup#2799]
- Rollup Version: 1.9.0
- Operating System (or Browser): osx and browser

### How Do We Reproduce?

I am not sure about the root of the problem, but we hit the issue when we try to treeshake part of the stencil's runtime using a object of constant booleans.

https://rollupjs.org/repl?version=1.9.0&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmNvbnN0JTIwQlVJTEQlMjAlM0QlMjB0cnVlJTNCJTVDbiU1Q25leHBvcnQlMjBjb25zdCUyMGJhciUyMCUzRCUyMCgpJTIwJTNEJTNFJTIwJTdCJTVDbiUyMCUyMCUyMCUyMGNvbnNvbGUubG9nKGZvbygpKSUzQiU1Q24lN0QlM0IlNUNuJTVDbmNvbnN0JTIwZm9vJTIwJTNEJTIwKCklMjAlM0QlM0UlMjAlN0IlNUNuJTVDdHJldHVybiUyMEJVSUxEJTIwJTNGJTIwQSUyMCUzQSUyMEIlM0IlNUNuJTdEJTNCJTIyJTJDJTIyaXNFbnRyeSUyMiUzQXRydWUlN0QlNUQlMkMlMjJvcHRpb25zJTIyJTNBJTdCJTIyZm9ybWF0JTIyJTNBJTIyZXNtJTIyJTJDJTIybmFtZSUyMiUzQSUyMm15QnVuZGxlJTIyJTJDJTIyYW1kJTIyJTNBJTdCJTIyaWQlMjIlM0ElMjIlMjIlN0QlMkMlMjJnbG9iYWxzJTIyJTNBJTdCJTdEJTdEJTJDJTIyZXhhbXBsZSUyMiUzQW51bGwlN0Q=

Replacing `BUILD` with `true` directly, removes the dead-code properly.

### Expected Behavior

**Input**:
```js
const BUILD = true;

export const bar = () => {
    console.log(foo());
};

const foo = () => {
	return BUILD ? A : B;
};
```

**Should generate**

```js
const bar = () => {
    console.log(foo());
};

const foo = () => {
	return A;
};

export { bar };
```

### Actual Behavior

```js
const BUILD = true;

const bar = () => {
    console.log(foo());
};

const foo = () => {
	return BUILD ? A : B;
};

export { bar };
```

The constant expression is not dropped

**Comment by lukastaegert** [comment]
For now  this is an ordering issue. If you switch the order of the definitions of `foo` and `bar`, it will work correctly. Will check if there is a simple way of improving it without introducing a second `bind` pass...

**Comment by lukastaegert** [comment]
The fix was surprisingly easy, see #2803
